v1_21_R2/src/main/java/net/citizensnpcs/nms/v1_21_R2/util/NMSImpl.java

```diff
@@ -2204,6 +2208,23 @@ private static Object deserialiseNBT(Tag tag) {
         throw new IllegalArgumentException();
     }
 
+    public static <T extends Entity & NPCHolder> void callNPCMoveEvent(T what) {
```

Comment:
Move this to NMS

Reply:
it is impossible as T is NMS entity.

Reply:
Just thinking if we should call PlayerMoveEvent here...

Reply:
It is techinally possible to add. just check if the T is a Player.

Reply:
But I'm not recommended to do this as event constructor is not for plugins